lib/improver/wxcode/weather_symbols.py

```diff
@@ -141,16 +156,28 @@ def check_input_cubes(self, cubes):
                         not self.coord_named_threshold):
                     self.coord_named_threshold = True
 
-                test_condition = (
-                    iris.Constraint(
-                        coord_values={threshold_name: lambda cell: (
-                            threshold * (1. - self.float_tolerance) < cell <
-                            threshold * (1. + self.float_tolerance))},
-                        cube_func=lambda cube: (
-                            find_threshold_coordinate(
-                                cube
-                            ).attributes['spp__relative_to_threshold'] ==
-                            condition)))
+                if threshold == 0.0:
```

Comment:
Also we don't want to check for equalities of floats in this way. Better to have something like `if abs(threshold) < 1e-5` or some other small tolerance.

Reply:
This doesn't work as the rainfall rates can be very small amounts.

Reply:
Presumably we know the units though, so can predict how small it could sensibly be? The units should be m s-1, so 1e-12 ought to be a good tolerance. Would this work?

Reply:
Possibly for the rainfall rate it would work but what if we had something else in SI units that was even smaller and then in the constraint it is set ==. Would we then set that to abs(threshold) < 1e-12. It is much harder to understand what this is trying to do using 1e-12. I know it is bad practice for to use equality with real numbers but pragmatically it is done all the time and seems to work fine. I will change this if you think it really is a major issue but I would prefer not to.

Reply:
If this is not uniquely for precipitation rate, then I agree that it is very difficult to avoid this direct comparison. Probably better to leave it alone.

Reply:
This is for number of lightning strikes per unit area, I'm not sure how small that goes.

Reply:
```
if abs(threshold) < 5*np.finfo(float).eps:
```
?

Reply:
(FYI, `np.finfo(float).eps` returns the epsilon, from the numpy documentation:
"The smallest representable positive number such that 1.0 + eps != 1.0. Type of eps is an appropriate floating point type.")

Reply:
Please tell me you are joking. Relying on some magic number in numpy seems madness. I've had a look at some lightning data. Mostly it is all zero but in a case where there was some lightning it didn't seem to go below 1e-10 so for now I think 1e-12 should be OK as a float_abs_tolerance threshold for threshold == 0.0. I doubt we would ever use an actual threshold value for lightning other than 0.0 anyway.